src/v/cluster/controller_backend.cc

```diff
@@ -651,6 +742,28 @@ ss::future<std::error_code> controller_backend::create_partition(
       .then([] { return make_error_code(errc::success); });
 }
 
+ss::future<std::error_code> controller_backend::shutdown_on_current_shard(
+  model::ntp ntp, model::revision_id rev) {
+    vlog(clusterlog.trace, "shutting down partition: {}", ntp);
+    auto partition = _partition_manager.local().get(ntp);
+    // partition doesn't exists it was deleted
+    if (!partition) {
+        co_return errc::partition_not_exists;
+    }
+    auto gr = partition->group();
```

Comment:
We need all of these co_await in a try & crash() since it is unrecoverable to have partial failure.

Reply:
@mmaslankaprv I think this is a good idea. it seems critical enough to crash over a single problem with this process.